src/languages.rs

```diff
@@ -1,151 +1,380 @@
+    #[strum(serialize = "Shona", serialize = "sn")]
     SN,
+    #[strum(serialize = "Sindhi", serialize = "sd")]
     SD,
+    #[strum(serialize = "Sinhala", serialize = "si")]
     SI,
-    SS,
+    #[strum(serialize = "Slovak", serialize = "sk")]
     SK,
+    #[strum(serialize = "Slovenian", serialize = "sl")]
     SL,
+    #[strum(serialize = "Somali", serialize = "so")]
     SO,
+    #[strum(serialize = "Southern Sotho", serialize = "st")]
+    ST,
+    #[strum(serialize = "Spanish", serialize = "es")]
     ES,
+    #[strum(serialize = "Sundanese", serialize = "su")]
     SU,
+    #[strum(serialize = "Swahili", serialize = "sw")]
     SW,
+    #[strum(serialize = "Swati", serialize = "ss")]
+    SS,
+    #[strum(serialize = "Swedish", serialize = "sv")]
     SV,
+    #[strum(serialize = "Tagalog", serialize = "tl")]
     TL,
+    #[strum(serialize = "Tahitian", serialize = "ty")]
+    TY,
+    #[strum(serialize = "Tajik", serialize = "tg")]
     TG,
+    #[strum(serialize = "Tamil", serialize = "ta")]
     TA,
+    #[strum(serialize = "Tatar", serialize = "tt")]
     TT,
+    #[strum(serialize = "Telugu", serialize = "te")]
     TE,
+    #[strum(serialize = "Thai", serialize = "th")]
     TH,
+    #[strum(serialize = "Tibetan", serialize = "bo")]
     BO,
+    #[strum(serialize = "Tigrinya", serialize = "ti")]
     TI,
+    #[strum(serialize = "Tonga", serialize = "to")]
     TO,
+    #[strum(serialize = "Tsonga", serialize = "ts")]
     TS,
+    #[strum(serialize = "Tswana", serialize = "tn")]
+    TN,
+    #[strum(serialize = "Turkish", serialize = "tr")]
     TR,
+    #[strum(serialize = "Turkmen", serialize = "tk")]
     TK,
+    #[strum(serialize = "Twi", serialize = "tw")]
     TW,
+    #[strum(serialize = "Uighur", serialize = "ug")]
     UG,
+    #[strum(serialize = "Ukrainian", serialize = "uk")]
     UK,
+    #[strum(serialize = "Urdu", serialize = "ur")]
     UR,
+    #[strum(serialize = "Uzbek", serialize = "uz")]
     UZ,
+    #[strum(serialize = "Venda", serialize = "ve")]
+    VE,
+    #[strum(serialize = "Vietnamese", serialize = "vi")]
     VI,
+    #[strum(serialize = "Volapük", serialize = "vo")]
     VO,
+    #[strum(serialize = "Walloon", serialize = "wa")]
+    WA,
+    #[strum(serialize = "Welsh", serialize = "cy")]
     CY,
+    #[strum(serialize = "Wolof", serialize = "wo")]
     WO,
+    #[strum(serialize = "Xhosa", serialize = "xh")]
     XH,
+    #[strum(serialize = "Sichuan Yi", serialize = "ii")]
+    II,
+    #[strum(serialize = "Yiddish", serialize = "yi")]
     YI,
+    #[strum(serialize = "Yoruba", serialize = "yo")]
     YO,
+    #[strum(serialize = "Zhuang", serialize = "za")]
     ZA,
-    ZU
+    #[strum(serialize = "Zulu", serialize = "zu")]
+    ZU,
 }
 
 impl Iso639a {
     pub fn is_valid(lang: &str) -> bool {
-        Self::languages()
-            .iter()
-            .any(|valid_lang| valid_lang == lang)
+        Iso639a::from_str(lang).is_ok()
     }
 
     pub fn languages() -> Vec<String> {
```

Comment:
Since we now have a Display this function could be removed and it's use could be replaced for something along the lines of `format!("{:?} ({:#})", ...)`.

So the error becomes "{} is not a valid ISO 639-1 language, valid languages include\n" and then a map of all the formats I specified above in this message.